cmd/goplugin-foobernetes/main.go

```diff
@@ -8,12 +8,19 @@ import (
 )
 
 func main() {
+	// Always set the Level to "Trace" and JSONFormat to "true" to allow Lyra to handle the log messages properly
 	hclog.DefaultOptions = &hclog.LoggerOptions{
-		Name:            "Puppet",
-		Level:           hclog.Debug,
+		Name:            "Foobernetes",
+		Level:           hclog.Trace,
```

Comment:
Doesn't this mean that the plugin will produce a lot of unnecessary output that might then just be discarded by lyra? I would like to see a solution where the level is passed on from Lyra to the plug-in. The reason for this is performance. It might be expensive to format log messages and sending all the data from the plug-in to Lyra is bad unless it's used. Lyra needs to parse the JSON to understand what it is and only after that, when it knows the level, can it discard it if the level isn't active.

Reply:
I would prefer that too. I don't think go-plugin supports that natively though - as far as I can see this is how the library is intended to be used. I would rather go ahead with this solution that gives us functionally correct logging then raise a separate issue to revisit improving the efficiency.

It's not the case for the identity and the puppet plugins, but in the case of cloud plugins I don't think the performance cost will be meaningful. Those are interacting with APIs out in the world and the cost of generating unneeded log messages seems negligible in comparison.

What do you think?

Reply:
Although... I've just an idea for something to try to let us do this out-of-band. Let me give it a shot tomorrow.

Reply:
https://github.com/lyraproj/servicesdk/pull/25 offers a possible solution.